### PR TITLE
[JIRA] TH2-1132 Implement bookmarks list

### PR DESCRIPTION
**Pull request by VekesserVladimir**

### CODE REVIEW

File: src/components/BookmarksPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+interface SavedItem {
+	value: EventMessage | EventAction;
+	type: 'pinned-message' | 'event';
+}
+
+const BookmarksPanel = () => {
+	const selectedStore = useSelectedStore();
+	const savedItems: SavedItem[] = React.useMemo(() => {
+		return ([...selectedStore.attachedMessages, ...selectedStore.pinnedMessages] as (
+			| EventAction
```
Comment (rzvdv) [inline src/components/BookmarksPanel.tsx:36]: why do we need this type casting?

File: src/components/BookmarksPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			| EventMessage
+		)[])
+			.map(item => ({
+				value: item,
+				type: isEventMessage(item) ? ('pinned-message' as 'pinned-message') : ('event' as 'event'),
```
Comment (rzvdv) [inline src/components/BookmarksPanel.tsx:41]: make a enum for selected items types

File: src/components/BookmarksPanel.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+			}))
+			.sort((firstItem, secondItem) => {
+				const getItemTimestamp = (item: SavedItem) =>
+					getTimestampAsNumber(
+						item.type === 'event'
+							? (item.value as EventAction).startTimestamp
```
Comment (rzvdv) [inline src/components/BookmarksPanel.tsx:47]: we have type guards for Events and Message, (isEventMessage, isEventAction) why don't we check value? either use it or create a new type guard based on item type

File: src/components/BookmarksPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+					item.type === 'event'
+						? (item.value as EventAction).successful
+							? 'passed'
+							: 'failed'
+						: null,
+				);
+				return (
+					<div
+						key={
+							item.type === 'event'
```
Comment (rzvdv) [inline src/components/BookmarksPanel.tsx:82]: same

File: src/stores/SelectedStore.ts [hunk 5]
```diff
@@ -77,6 +77,14 @@ export class SelectedStore {
 		localStorageWorker.setPersistedPinnedMessages(this.pinnedMessages);
 	};
 
+	@action
+	public removePinnedMessageFromLocalStorage = (removedMessage: EventMessage) => {
```
Comment (rzvdv) [inline src/stores/SelectedStore.ts:81]: maybe we'll just have a reaction on pinnedMessagesChanges which updaes pinned messages to localStorage?

File: src/util/LocalStorageWorker.ts [hunk 6]
```diff
@@ -16,18 +16,36 @@
 
 import { EventMessage } from '../models/EventMessage';
 
+const persistedPinnedMessageKeyPart = 'pinnedMessage';
 class LocalStorageWorker {
 	getPersistedPinnedMessages(): EventMessage[] {
 		try {
-			const stringPersistedPinnedMessages = localStorage.getItem('pinnedMessages');
-			return stringPersistedPinnedMessages ? JSON.parse(stringPersistedPinnedMessages) : [];
+			const persistedPinnedMessages: EventMessage[] = [];
```
Comment (rzvdv) [inline src/util/LocalStorageWorker.ts:23]: i don't like idea of keeping every pinned message separately, why don't we just store them inside of collection?

**Review by rzvdv — CHANGES_REQUESTED** [review]

File: src/components/BookmarksPanel.tsx (hunk 1 above)
Comment (VekesserVladimir) [reply src/components/BookmarksPanel.tsx:36]: We need this casting until we get pinned messages functionality. It shows that the array contains events and messages

Comment (rzvdv) [reply src/components/BookmarksPanel.tsx:36]: i still don't get it. either way you can create observable pinnedEvents, initialize it with empty array and add functionality later

**Review by rzvdv — APPROVED** [review]

**Pull request merged**